Userland/Libraries/LibDesktop/AppFile.h

```diff
@@ -32,6 +32,7 @@ class AppFile : public RefCounted<AppFile> {
     String icon_path() const;
     GUI::Icon icon() const;
     bool run_in_terminal() const;
+    bool requires_root() const;
```

Comment:
This feels like three separate  and distinct changes, but it could still be 2. The first one being updating AppFile to have this new parameter, and the second/third being updating Settings to respect the option, and marking NetworkSettings as a root-requiring process.